### ISSUE TITLE
You have provided a history object created with history v4.x or v2.x and earlier.

### ISSUE THREAD

**Issue by dalbinaco** [pedronauck/docz#266]
## Bug Report

I'm not able to access the documentation generated by docz. It compiles without any errors but when opening the browser to check out the pages I get this:

![image](https://user-images.githubusercontent.com/16964673/44804848-fe9ed180-ab98-11e8-8e89-518e35e94915.png)

```
You have provided a history object created with history v4.x or v2.x and earlier. This version of React Router is only compatible with v3 history objects. Please change to history v3.x.
```

**To Reproduce**

1. Run docz on development mode
2. Open `localhost:3000` in browser

```shell
You have provided a history object created with history v4.x or v2.x and earlier. This version of React Router is only compatible with v3 history objects. Please change to history v3.x.
Stack trace

in Router (created by BrowserRouter)
in BrowserRouter (created by DoczTheme)
in DataServer (created by DoczTheme)
in CopyOnWriteStoreProvider (created by DoczTheme)
in ErrorBoundary (created by DoczTheme)
in DoczTheme (at root.jsx:7)
in Root (created by HotExportedRoot)
in AppContainer (created by HotExportedRoot)
in HotExportedRoot (at index.jsx:17)
```

**`package.json`**
```json
"devDependencies": {
    "docz": "^0.10.3",
    "eslint-config-airbnb": "^16.1.0",
    "eslint-plugin-import": "^2.12.0",
    "eslint-plugin-jsx-a11y": "^5.0.1",
    "eslint-plugin-react": "^7.8.2",
    "gulp": "^3.9.1",
    "gulp-inject": "^4.3.2",
    "husky": "^0.14.3",
    "lint-staged": "^7.2.0",
    "react-scripts": "1.1.4",
    "webpack": "^4.17.1"
  },
  "dependencies": {
    "axios": "^0.17.1",
    "bootstrap": "^4.1.1",
    "classnames": "^2.2.5",
    "css-element-queries": "^1.0.2",
    "graphql": "^0.13.2",
    "graphql-tag": "^2.9.2",
    "gsap": "^1.20.3",
    "history": "^4.7.2",
    "localforage": "^1.5.6",
    "lodash": "^4.17.10",
    "moment": "^2.22.1",
    "moment-duration-format": "^2.2.2",
    "node-sass-chokidar": "^1.3.0",
    "npm-run-all": "^4.1.3",
    "paper": "^0.11.5",
    "prop-types": "^15.6.1",
    "query-string": "^5.0.1",
    "rc-slider": "^8.6.1",
    "rc-steps": "^3.1.1",
    "react": "^16.4.0",
    "react-apollo": "^2.1.8",
    "react-app-rewire-hot-loader": "^1.0.1",
    "react-app-rewired": "^1.5.2",
    "react-copy-to-clipboard": "^5.0.1",
    "react-datepicker": "^1.4.1",
    "react-dom": "^16.4.0",
    "react-feather": "^1.1.0",
    "react-helmet": "^5.2.0",
    "react-hot-loader": "^4.3.0",
    "react-infinite-scroller": "^1.1.3",
    "react-player": "^1.3.1",
    "react-redux": "^5.0.6",
    "react-router": "^3.2.0",
    "react-router-redux": "^4.0.8",
    "react-select-plus": "^1.2.0",
    "react-tabs": "^2.2.2",
    "react-toggle": "^4.0.2",
    "react-toggle-switch": "^3.0.4",
    "reactstrap": "^6.3.1",
    "redux": "^3.7.2",
    "redux-action-buffer": "^1.2.0",
    "redux-logger": "^3.0.6",
    "redux-persist": "^5.5.0",
    "redux-promise": "^0.6.0",
    "redux-thunk": "^2.2.0",
    "shortid": "^2.2.8",
    "styled-bootstrap-grid": "^1.0.4",
    "styled-components": "^3.3.3",
    "styled-system": "^3.0.2",
    "styled-theming": "^2.2.0",
    "underscore": "^1.9.1",
    "whatwg-fetch": "^2.0.3"
  },
  "scripts": {
    "gulp": "gulp",
    "build-css": "node-sass-chokidar --include-path ./src --include-path ./node_modules src/ -o src/",
    "watch-css": "npm run build-css && node-sass-chokidar --include-path ./src --include-path ./node_modules src/ -o src/ --watch --recursive",
    "start-js": "react-app-rewired start",
    "start": "npm-run-all -p watch-css start-js",
    "build-js": "react-app-rewired build",
    "build": "npm-run-all build-css build-js gulp",
    "test": "react-app-rewired test --env=jsdom",
    "eject": "react-app-rewired eject",
    "precommit": "lint-staged",
    "lint:check": "eslint src",
    "lint:fix": "eslint --fix src",
    "docz:dev": "docz dev"
  },
  "lint-staged": {
    "*.js": [
      "eslint"
    ]
  }
```

**Expected behavior**

Not crash

**Environment**

* OS: elementaryOS 0.4.1
* Node/npm version: Node v10.0.0 / npm v6.1.0

**Comment by sallas** [comment]
The problem is that you are using incompatible versions between "react-router" and "history".

Attempt to update "react-router" to 4.3 and see if this helps you.

**Comment by dalbinaco** [comment]
@sallas I updated `react-router` to `4.3.1` and it works now, thank you!

**Comment by vivek12345** [comment]
@pedronauck  I have been using react-router v3 in my application and I do not wish to upgrade to v4. Any solution to this problem without upgrading to v4.

Here is my package.json file
```
{
  "name": "client",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@babel/core": "^7.0.0",
    "@babel/plugin-proposal-class-properties": "^7.0.0",
    "@babel/plugin-proposal-object-rest-spread": "^7.0.0",
    "@babel/plugin-syntax-dynamic-import": "^7.0.0",
    "@babel/plugin-transform-runtime": "^7.0.0",
    "@babel/preset-env": "^7.0.0",
    "@babel/preset-react": "^7.0.0",
    "@babel/runtime": "^7.0.0",
    "autoprefixer": "7.1.6",
    "axios": "^0.18.0",
    "babel-core": "^7.0.0-bridge.0",
    "babel-eslint": "7.2.3",
    "babel-jest": "20.0.3",
    "babel-loader": "^8.0.0",
    "babel-preset-react-app": "^3.1.1",
    "case-sensitive-paths-webpack-plugin": "2.1.1",
    "chalk": "1.1.3",
    "classnames": "^2.2.6",
    "css-loader": "0.28.7",
    "dotenv": "4.0.0",
    "dotenv-expand": "4.2.0",
    "eslint": "4.10.0",
    "eslint-loader": "1.9.0",
    "eslint-plugin-flowtype": "2.39.1",
    "eslint-plugin-import": "2.8.0",
    "eslint-plugin-jsx-a11y": "5.1.1",
    "eslint-plugin-react": "7.4.0",
    "extract-text-webpack-plugin": "3.0.2",
    "file-loader": "1.1.5",
    "fs-extra": "3.0.1",
    "html-webpack-plugin": "2.29.0",
    "i18next": "^11.6.0",
    "i18next-browser-languagedetector": "^2.2.3",
    "i18next-xhr-backend": "^1.5.1",
    "jest": "20.0.4",
    "lodash": "^4.17.10",
    "moment": "^2.22.2",
    "object-assign": "4.1.1",
    "postcss-flexbugs-fixes": "3.2.0",
    "postcss-loader": "2.0.8",
    "promise": "8.0.1",
    "raf": "3.4.0",
    "react": "^16.4.2",
    "react-dev-utils": "^5.0.1",
    "react-dom": "^16.4.2",
    "react-i18next": "^7.10.1",
    "react-redux": "^5.0.7",
    "react-router": "3.2.0",
    "redux": "^4.0.0",
    "redux-thunk": "^2.3.0",
    "resolve": "1.6.0",
    "style-loader": "0.19.0",
    "sw-precache-webpack-plugin": "0.11.4",
    "url-loader": "0.6.2",
    "webpack": "3.8.1",
    "webpack-dev-server": "2.9.4",
    "webpack-manifest-plugin": "1.3.2",
    "whatwg-fetch": "2.0.3",
    "yup": "^0.26.2"
  },
  "scripts": {
    "start": "node scripts/start.js",
    "build": "CI=false node scripts/build.js",
    "test": "CI=false node scripts/test.js --env=jsdom",
    "test:watch": "node scripts/test.js --env=jsdom --watch",
    "test:coverage": "node scripts/test.js --env=jsdom --coverage",
    "docz:dev": "docz dev",
    "docz:build": "docz build",
    "eslint:fix": "eslint --ignore-path .gitignore --fix"
  },
  "lint-staged": {
    "src/**/*.{js,jsx}": [
      "npm run eslint:fix",
      "git add"
    ],
    "src/**/*.{js,jsx,json,css}": [
      "prettier --config .prettierrc --write",
      "git add"
    ]
  },
  "jest": {
    "collectCoverageFrom": [
      "src/**/*.{js,jsx,mjs}",
      "!src/i18n/*.{js,jsx}",
      "!src/index.js"
    ],
    "setupFiles": [
      "<rootDir>/config/polyfills.js",
      "<rootDir>/config/jest/jestSetup.js",
      "<rootDir>/src/__mocks__/browserMocks.js"
    ],
    "testMatch": [
      "<rootDir>/src/**/__tests__/**/*.{js,jsx,mjs}",
      "<rootDir>/src/**/?(*.)(spec|test).{js,jsx,mjs}"
    ],
    "testEnvironment": "node",
    "testURL": "http://localhost",
    "transform": {
      "^.+\\.(js|jsx|mjs)$": "<rootDir>/node_modules/babel-jest",
      "^.+\\.css$": "<rootDir>/config/jest/cssTransform.js",
      "^(?!.*\\.(js|jsx|mjs|css|json)$)": "<rootDir>/config/jest/fileTransform.js"
    },
    "transformIgnorePatterns": [
      "[/\\\\]node_modules[/\\\\].+\\.(js|jsx|mjs)$"
    ],
    "moduleNameMapper": {
      "^react-native$": "react-native-web",
      "@helpers(.*)$": "<rootDir>/src/helpers/$1",
      "@constants(.*)$": "<rootDir>/src/constants/$1",
      "@components(.*)$": "<rootDir>/src/components/$1",
      "@pages(.*)$": "<rootDir>/src/pages/$1",
      "@apis(.*)$": "<rootDir>/src/apis/$1",
      "@actions(.*)$": "<rootDir>/src/actions/$1",
      "@src(.*)$": "<rootDir>/src/$1"
    },
    "moduleFileExtensions": [
      "web.js",
      "js",
      "json",
      "web.jsx",
      "jsx",
      "node",
      "mjs"
    ]
  },
  "devDependencies": {
    "docz": "^0.10.3",
    "enzyme": "^3.4.1",
    "enzyme-adapter-react-16": "^1.2.0",
    "formik": "^1.0.2",
    "redux-mock-store": "^1.5.3"
  }
}
```

**Comment by pedronauck** [comment]
I'll put react-router inside the bundle to avoid this type of incompatibility @vivek12345 👍
